src/openrct2/scenario/ScenarioRepository.cpp

```diff
@@ -272,6 +275,36 @@ class ScenarioRepository final : public IScenarioRepository
         delete scanner;
     }
 
+    void ScanMegaPark(std::string &mpdatDir, std::string &scenarioDir)
+    {
+        //Convert mp.dat from RCT1 Data directory into SC21.sc4 (Mega Park)
+        utf8 mpdatPath[MAX_PATH];
+        utf8 sc21Path[MAX_PATH];
+
+        String::Set(mpdatPath, sizeof(mpdatPath), mpdatDir.c_str());
+
+        String::Set(sc21Path, sizeof(sc21Path), scenarioDir.c_str());
+        //Ensure the scenario directory exists first
+        platform_ensure_directory_exists(sc21Path); 
```

Comment:
This call should only take place if we find `mp.dat` and have something to convert.